src/workloads/query/TimeseriesTSBSOptimizations.yml

```diff
@@ -0,0 +1,286 @@
+# return empty documents. The following task tests this optimization.
+  - Name: EmptyUnpackWithGroupSum
+    Type: CrudActor
+    Database: *database
+    Threads: 10
+    Phases:
+      OnlyActiveInPhases:
+        Active: [3]
+        NopInPhasesUpTo: *maxPhases
+        PhaseConfig:
+          Repeat: *repeat
+          Database: *database
+          Collection: *collection
+          Operations:
+            - OperationMetricsName:
+              OperationName: aggregate
+              OperationCommand:
+                Pipeline: [
+                  { 
+                    $group: 
+                    { 
+                      _id: 5,
+                      new_count: {$sum: 2},
+                    }
+                  }
+                ]
+
+# A new optimization for $count was added in 7.1 Below are tasks that test the $count stage and accumulator
+# optimization. We expect these tasks to run faster on version >=7.1.
+  - Name: EmptyUnpackWithCountStage
+    Type: CrudActor
+    Database: *database
+    Threads: 10
+    Phases:
+      OnlyActiveInPhases:
+        Active: [4]
+        NopInPhasesUpTo: *maxPhases
+        PhaseConfig:
+          Repeat: *repeat
+          Database: *database
+          Collection: *collection
+          Operations:
+            - OperationMetricsName:
+              OperationName: aggregate
+              OperationCommand:
+                Pipeline:  [{ $count: "count" }]
+
+  - Name: GroupRewriteCount
+    Type: CrudActor
+    Database: *database
+    Threads: 10
+    Phases:
+      OnlyActiveInPhases:
+        Active: [5]
+        NopInPhasesUpTo: *maxPhases
+        PhaseConfig:
+          Repeat: *repeat
+          Database: *database
+          Collection: *collection
+          Operations:
+            - OperationMetricsName:
+              OperationName: aggregate
+              OperationCommand:
+                Pipeline: [
+                  { 
+                    $group: 
+                    { 
+                      _id: "$tags.hostname",  
+                      document_count: { $count: {} }
+                    }
+                  }
+                ]
+
+# Predicates generated for a $match on the timeField are not applied to the unpacked documents if the entire bucket
+# matches the predicate. This task tests that optimization.
+  - Name: WholeBucketFilter
+    Type: CrudActor
+    Database: *database
+    Threads: 10
+    Phases:
+      OnlyActiveInPhases:
+        Active: [6]
+        NopInPhasesUpTo: *maxPhases
+        PhaseConfig:
+          Repeat: *repeat
+          Database: *database
+          Collection: *collection
+          Operations:
+            - OperationMetricsName:
+              OperationName: aggregate
+              OperationCommand:
+                Pipeline:  [ 
+                  { 
+                    $match: { "time": { $lt: {^Date: "2016-01-15T01:49:23"} } } 
```

Comment:
Could you please add a comment re how many buckets this filter hits? (for the optimization to becomes noticeable, the number of hit buckets should be relatively high)